src/administration/guards/admin.guard.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class AdminAuthorizationGuard implements CanActivate {
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
+    if (request.user && request.user.profile === 'admin') {
```

Comment:
Suggetion: make practice of use enums for this type of static strings, it will more helpful for future project